CIP-XXXX/README.md

```diff
@@ -0,0 +1,87 @@
+---
+CIP: XXXX
+Title: Non-segregated Block Body Serialization
+Category: Ledger
+Authors:
+  - Teodora Danciu <teodora.danciu@iohk.io>
+  - Alexey Kuleshevich <alexey.kuleshevich@iohk.io>
+Implementors: N/A
+Status: Proposed
+Discussions:
+  - https://github.com/IntersectMBO/cardano-ledger/issues/5046
+  - https://github.com/cardano-foundation/CIPs/pull/1084
+Created: 2025-07-30
+License: CC-BY-4.0
+---
+
+## Abstract
+
+We propose changing the CBOR encoding of a block body from a segregated layout to a plain sequence of transactions.
+Current layout: all transaction bodies are concatenated and encoded first, followed by their witness sets, then followed by auxiliary-data hashes, and finally followed by validity flags.
+Proposed layout: each transaction is serialized in full before the next transaction is written to the stream.
+
+## Motivation: why is this CIP necessary?
+
+Segregated serialization of [Nested transactions](https://github.com/cardano-foundation/CIPs/pull/862) would be challenging both to specify and implement.
+Separating and concatenating components across nested and non-nested transactions introduces complexity that is error-prone and potentially inefficient, as it may require tracking offsets and performing additional buffering and copying at runtime.
+
```

Comment:
This CIP will also slightly simplify implementation of #964 as it requires each transaction bytes to be hashed. However this is always possible to do regardless of the format, so this CIP is not a strict dependency for #964.